Modules/MSCloudLoginAssistant/MSCloudLoginAssistant.psm1

```diff
@@ -263,74 +284,202 @@ function Connect-M365Tenant
-            $Global:MSCloudLoginConnectionProfile.Tasks.CertificatePath = $CertificatePath
-            $Global:MSCloudLoginConnectionProfile.Tasks.CertificatePassword = $CertificatePassword
-            $Global:MSCloudLoginConnectionProfile.Tasks.AccessTokens = $AccessTokens
-            $Global:MSCloudLoginConnectionProfile.Tasks.Endpoints = $Endpoints
-            $Global:MSCloudLoginConnectionProfile.Tasks.Connect()
+            $Script:MSCloudLoginConnectionProfile.Tasks.Credentials = $Credential
+            $Script:MSCloudLoginConnectionProfile.Tasks.ApplicationId = $ApplicationId
+            $Script:MSCloudLoginConnectionProfile.Tasks.ApplicationSecret = $ApplicationSecret
+            $Script:MSCloudLoginConnectionProfile.Tasks.TenantId = $TenantId
+            $Script:MSCloudLoginConnectionProfile.Tasks.CertificateThumbprint = $CertificateThumbprint
+            $Script:MSCloudLoginConnectionProfile.Tasks.CertificatePath = $CertificatePath
+            $Script:MSCloudLoginConnectionProfile.Tasks.CertificatePassword = $CertificatePassword
+            $Script:MSCloudLoginConnectionProfile.Tasks.AccessTokens = $AccessTokens
+            $Script:MSCloudLoginConnectionProfile.Tasks.Endpoints = $Endpoints
+            $Script:MSCloudLoginConnectionProfile.Tasks.Connect()
         }
     }
 }
 
+<#
+.SYNOPSIS
+    This function returns the connection profile for a specific workload.
+.DESCRIPTION
+    This function returns the connection profile for a specific workload. A caller can use this function to get connection information for a specific workload.
+.OUTPUTS
+    Object (or $null). Get-MSCloudLoginConnectionProfile returns the connection profile for a specific workload or $null, if no connection profile exists.
+.EXAMPLE
+    Get-MSCloudLoginConnectionProfile -Workload 'MicrosoftGraph'
+#>
+function Get-MSCloudLoginConnectionProfile
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [ValidateSet('AdminAPI', 'Azure','AzureDevOPS', 'ExchangeOnline', 'Fabric', `
+                'SecurityComplianceCenter', 'PnP', 'PowerPlatforms', `
+                'MicrosoftTeams', 'MicrosoftGraph', 'SharePointOnlineREST', 'Tasks', 'DefenderForEndpoint')]
+        [System.String]
+        $Workload
+    )
+
+    if ($null -ne $Script:MSCloudLoginConnectionProfile.$Workload)
+    {
+        return $Script:MSCloudLoginConnectionProfile.$Workload.Clone()
+    }
+}
+
+<#
+.SYNOPSIS
+    This function resets the entire connection profile.
+.DESCRIPTION
+    This function resets the entire connection profile. It is used to disconnect all workloads and reset the connection profile.
+.EXAMPLE
+    Reset-MSCloudLoginConnectionProfileContext
+#>
+function Reset-MSCloudLoginConnectionProfileContext
+{
+    $source = 'Reset-MSCloudLoginConnectionProfileContext'
+    Add-MSCloudLoginAssistantEvent -Message 'Resetting connection profile' -Source $source
```

Comment:
Shouldn't we add some logic to disconnect any open sessions? That way we clean up everything that we have created. Is also one of the Exchange best practices.

Pseudo code:
If ConnectedToExchange Then DisconnectExchange
If ConnectedToGraph Then DisconnectGraph
Etc, etc